tests/loops/test_loops.py

```diff
@@ -731,7 +732,116 @@ def test_fit_loop_reset(tmpdir):
     assert epoch_loop.restarting
     assert epoch_loop.batch_progress.total.ready == 4
     assert epoch_loop.batch_progress.total.completed == 3  # the checkpoint was saved on train_batch_end
-    assert epoch_loop.batch_progress.current.ready == 0
-    assert epoch_loop.batch_progress.current.completed == 0
+    assert epoch_loop.batch_progress.current.ready == 4
+    assert epoch_loop.batch_progress.current.completed == 4
 
     assert optimizer_loop.optim_progress.optimizer_position == 1
+
+
+@mock.patch.dict(os.environ, {"PL_FAULT_TOLERANT_TRAINING": "1"})
+@RunIf(min_torch="1.7.0")
+@pytest.mark.parametrize(
+    "dataset_classes",
+    [
+        [[RandomGetItemDataset], [RandomGetItemDataset]],
+        [[RandomGetItemDataset], [RandomGetItemDataset, RandomGetItemDataset]],
+    ],
```

Comment:
the test is not dealing with randomness so we don't need to import these datasets from the other test suite. it would be saver not to in case we change them in the future. we could just use the regular random dataset that the boring model normally uses.

Reply:
Yes, done !